File: runtime/r2-technique/python/procedure-1/sample-application/malicious_dependency/foo.py
# Define a function in the dependency, that we suppose is the popular
# one for which this 3rd-party dependency is tipically downloaded
# by downstream users

def say_hello(name):
    import os; os.system("touch Hacked.txt")     # Place any malicious code :)
    return f"Hello, {name}!"

File: runtime/r3-technique/java/procedure-1/src/dependency/Foo.java

package dependency;



public class Foo
{
    private String name;

    public Foo(String name)
    {
        try{
            ProcessBuilder builder = new ProcessBuilder();
            builder.command("touch", "/tmp/Hacked.txt");
            Process p = builder.start();
        }catch(Exception e ){
            // Suppress warning
        }

        this.name = name;
    }

    public String sayHello() {
        return "Hello from " + name + "!";
    }
    
}
